be/src/exec/pipeline/analysis/analytic_sink_operator.cpp

```diff
@@ -12,7 +12,31 @@ Status AnalyticSinkOperator::prepare(RuntimeState* state) {
     // we must only prepare it at sink operator
     RETURN_IF_ERROR(
             _analytor->prepare(state, state->obj_pool(), get_memtracker(), get_memtracker(), get_runtime_profile()));
-    return _analytor->open(state);
+    RETURN_IF_ERROR(_analytor->open(state));
+
+    TAnalyticWindow window = _tnode.analytic_node.window;
+
+    if (!_tnode.analytic_node.__isset.window) {
+        _partition_finish = &AnalyticSinkOperator::_partition_finish_for_unbounded_frame;
+    } else if (window.type == TAnalyticWindowType::RANGE) {
+        // RANGE windows must have UNBOUNDED PRECEDING
+        // RANGE window end bound must be CURRENT ROW or UNBOUNDED FOLLOWING
+        if (!window.__isset.window_end) {
+            _partition_finish = &AnalyticSinkOperator::_partition_finish_for_unbounded_frame;
+        } else {
+            _partition_finish = &AnalyticSinkOperator::_partition_finish_for_unbounded_preceding_range_frame;
+        }
+    } else {
+        if (!window.__isset.window_start && !window.__isset.window_end) {
+            _partition_finish = &AnalyticSinkOperator::_partition_finish_for_unbounded_frame;
+        } else if (!window.__isset.window_start && window.window_end.type == TAnalyticWindowBoundaryType::CURRENT_ROW) {
+            _partition_finish = &AnalyticSinkOperator::_partition_finish_for_unbounded_preceding_rows_frame;
+        } else {
+            _partition_finish = &AnalyticSinkOperator::_partition_finish_for_sliding_frame;
```

Comment:
What's the meaning of  `_partition_finish`?

Reply:
It means  that the processing cursor has reached the end of the current partition